### DISCUSSION TITLE
Reuse Slingplayer Desktop after shutdown

### DISCUSSION THREAD

**Discussion by rbf121** [GerryDazoo/Slinger#92] [Ideas]
This may be more trouble then it's worth but I found that the sling player desktop software has a app preference defined which lets you set the starturl which the desktop app launches.

You can find it if you navigate to:
%appdata%\Sling Media\Slingplayer Desktop\Resources\AppPreference.ini

I navigated to the starturl that was defined there and downloaded the entire website, it's an html page and a few jquery ui .js files. Some interesting stuff in there, the .js file is 60k lines of code. Found this, which confirms things from another thread

        DEFAULT_ADMIN_PASSWORD: "admin",

And the launch URL

    FALCON: {        LAUNCH_URL: "http://newwatch.slingbox.com",        LAUNCH_URL_TO_CONNECT_TO_BOX: "http://newwatch.slingbox.com/connect?finderid=",        LAUNCH_URL_FOR_LANDING_PAGE: "http://newwatch.slingbox.com/?ticket="    },

I wonder if it's possible to set this code up on a dummy webserver, maybe even part of GerryDazoo's slinger app, to host this code and modify it however we need it to get through a configuration (including to change the launch URL). Then update the starturl to point to the new webserver to keep the desktop app functional.

Also, there is a console.log from the desktop app which has some interesting things in it, including the adminPassword. Which may be helpful for those that did not retrieve the adminPassword before servers shut down, but they have used the desktop app before in the past.

%appdata%\Sling Media\Slingplayer Desktop\Logs\console.log

The console log also listed out the IR codes when you connect to the device.

And there is a folder for remote codes which have been used on the machine before, including some image files,
%appdata%\Sling Media\WebSlingPlayer\Remote

**Comment by GerryDazoo** [comment]
This is great info. Thanks for digging into this. I think it would be easy to serve out that java script from my slingplayer app. You can easily redirect that url locally by editing the "hosts" file on your PC/Linux box. It might be lmhosts on the PC. 

- Gerry

PS. would you be interested in a preview of the 3.08 release? My testing has stalled due to a currently 4 day power outage where my slingboxes are located in Nova Scotia.

**Comment by rbf121** [comment]
Sure, I would give it a try.  There were a couple changes that I made to get things working for the 500 and I was going to wait and see if the next version of your software fixes.

**Comment by GerryDazoo** [comment]
I would really like to get a hold of the java script (or the whole shooting match). I've been discussing issues with reconfiguring the slingbox after Nov 9 on Reddit
https://www.reddit.com/r/slingbox/comments/y0y5m0/changing_settings_after_november_9/
I'm hoping the java script has the magic in it I need to properly authenticate an HTTP request so I can set passwords and upload remote control data files.

**Comment by rbf121** [comment]
Yea, that all seems to be in the javascript file.  You would need to do a bit of reverse engineering, the method calls are generic so might not be easiest to follow.

`        function y() {
            d.debug("inside _generateDigest()...");
            var a4 = N();
            a4 = a4 || "";
            var a2 = C() + ":" + a4 + ":" + v() + ":" + aA() + ":" + aP();
            var a3 = b.MD5(a2);
            d.debug("md5 hash ..." + a3);
            d.debug("exit _generateDigest()...");
            return a3
        }

        function ay() {
            d.debug("inside _getSlingAuthorizationHeader()...");
            s();
            f();
            var a2 = "account=" + C() + ", counter=" + aA() + ", cnonce=" + v() + ", digest=" + y();
            d.debug("value of Sling-Authorization - " + a2);
            d.debug("exit _getSlingAuthorizationHeader()...");
            return {
                "Sling-Authorization": a2
            }
        }
        function X() {
            d.debug("inside _getSlingAuthorizationQueryString()...");
            s();
            f();
            var a2 = "account=" + C() + "&counter=" + aA() + "&cnonce=" + v() + "&digest=" + y();
            d.debug("value of Sling-Authorization - " + a2);
            d.debug("exit _getSlingAuthorizationQueryString()...");
            return a2
        }`

Seems this function call is used a lot to build the URL you listed
`        function D() {
            return "http://" + q() + ":" + aG() + "/slingbox"
        }`

**Reply by GerryDazoo** [reply]
That's exactly what I was looking for.
